documentation/MSBuild-Server.md

```diff
@@ -0,0 +1,54 @@
+# MSBuild Server
+
+MSBuild Server is basically an another type of node which can accept build request from clients and utilize worker nodes in current fashion to build projects. Main purpose of the server node is to avoid expensive MSBuild process start during build from tools like .NET SDK.
+
+## Communication protocol
+
+The server node uses same IPC approach as current worker nodes - named pipes. This solution allows to reuse existing code. When process starts, pipe with deterministic name is opened and waiting for commands. Client has following worfklow:
```

Comment:
Using named pipes makes sense to me, but should this introduce a different serialization technology?

Reply:
We shall refrain from `BinaryFormatter` thats for sure. But since console redirection could be chatty, I see benefits in performance given by  custom written binary serialization format. Since we already have one which we use for communication between nodes, I would use it. 
Using some modern serialization frameworks like gRPC sounds appealing, but it will bring additional dependency.